core/src/main/java/com/linecorp/armeria/server/logging/AccessLogComponent.java

```diff
@@ -281,10 +364,101 @@ Object getMessage0(RequestLog log) {
             final Attribute<?> value = log.context().attr(key);
             return value != null ? stringifer.apply(value.get()) : null;
         }
+    }
+
+    /**
+     * A {@link RequestLog} component of a log message.
+     */
+    class RequestLogComponent extends ResponseHeaderConditional {
+
+        static boolean isSupported(AccessLogType type) {
+            return type == AccessLogType.REQUEST_LOG;
+        }
+
+        private final Function<RequestLog, Object> resolver;
+
+        RequestLogComponent(String variable, boolean addQuote,
+                            @Nullable Function<HttpHeaders, Boolean> condition) {
+            super(condition, addQuote);
+            resolver = findResolver(requireNonNull(variable, "variable"));
+        }
 
+        @Nullable
         @Override
-        public boolean addQuote() {
-            return addQuote;
+        Object getMessage0(RequestLog log) {
+            return resolver.apply(log);
+        }
+
+        @Nullable
+        private static String handleThrowable(@Nullable Throwable cause) {
+            if (cause == null) {
+                return null;
+            }
+            cause = Exceptions.peel(cause);
+            final String message = cause.getMessage();
+            return message != null ? cause.getClass().getSimpleName() + ": " + message
+                                   : cause.getClass().getSimpleName();
+        }
+
+        private static Function<RequestLog, Object> findResolver(String variable) {
```

Comment:
I don't know if it's a good idea or not, but how about using a `Map<String, Function<RequestLog, Object>>` here? 
If you think it takes too much memory and don't worth, just ignore this comment. :-)

Reply:
Thanks! This method is called by `ServerBuilder` which means that it's only at start-up. So I think that it's better to let it as it is. :-)